src/connectors/queue/stripe.ts

```diff
@@ -0,0 +1,76 @@
+import Queue from 'bull'
+
+import { QUEUE_JOB, QUEUE_NAME, QUEUE_PRIORITY } from 'common/enums'
+import logger from 'common/logger'
+import { PaymentService } from 'connectors'
+import SlackService from 'connectors/slack'
+
+import { BaseQueue } from './baseQueue'
+
+class StripeQueue extends BaseQueue {
+  paymentService: InstanceType<typeof PaymentService>
+  slackService: InstanceType<typeof SlackService>
+
+  constructor() {
+    super(QUEUE_NAME.stripe)
+    this.paymentService = new PaymentService()
+    this.slackService = new SlackService()
+    this.addConsumers()
+  }
+
+  /**
+   * Producers
+   */
+  addRepeatJobs = async () => {
+    // daily sync delivery failed event at 10:00
+    this.q.add(
+      QUEUE_JOB.syncDeliveryFailedEvents,
+      {},
+      {
+        priority: QUEUE_PRIORITY.MEDIUM,
+        repeat: { cron: '0 10 * * *', tz: 'Asia/Hong_Kong' },
+      }
+    )
+  }
+
+  /**
+   * Consumers
+   */
+  private addConsumers = () => {
+    this.q.process(
+      QUEUE_JOB.syncDeliveryFailedEvents,
+      this.syncDeliveryFailedEvents
+    )
+  }
+
+  private syncDeliveryFailedEvents: Queue.ProcessCallbackFunction<
+    unknown
+  > = async (job, done) => {
+    try {
+      logger.info('[schedule job] sync delivery failed events')
+
+      // query delivery failed events
+      const result = await this.paymentService.stripe.getDeliveryFailedEvents()
```

Comment:
Only retrieving events for the past 24 hours looks more polite if failed events can't be resolved on time.

Reply:
Adjusted the interval of repeat job from daily to 3hrs